Calendar/Get-CalendarDiagnosticObjectsSummary.ps1

```diff
@@ -59,19 +154,55 @@ function FindMatch {
     }
 }
 
+<#
+.SYNOPSIS
+Get the Mailbox for the Passed in Identity.
+Might want to extend to do 'Get-MailUser' as well.
+.PARAMETER CN of the Mailbox
+    The mailbox for which to retrieve properties.
+.PARAMETER Organization
+    [Optional] Organization to search for the mailbox in.
+#>
 function GetMailbox {
     param(
         [string]$Identity,
         [string]$Organization
     )
 
-    if ($Identity -and $Organization) {
-        $script:GetMailboxOutput = Get-Mailbox -Identity $Identity -Organization $Organization  -ErrorAction stop;
-        return $GetMailboxOutput;
-    }
-    if ($Identity -and !$Organization) {
-        $script:GetMailboxOutput = Get-Mailbox -Identity $Identity;
+    try {
+        Write-Output "Searching Get-Mailbox $(if ($Organization -ne `"`" ) {"with Org: $Organization"}) for $Identity."
+
+        # See if it is a Customer Tenant running the cmdlet. (They will not have access to Organization parameter)
+        $MSSupport = [Bool](Get-Help Get-Mailbox -Parameter Organization -ErrorAction SilentlyContinue)
+        if ($debug) { Write-Host -ForegroundColor Green "MSSupport: $MSSupport" }
+
+        if ($Identity -and $Organization) {
+            if ($MSSupport) {
+                if ($debug) { Write-Host -ForegroundColor Green "Using Organization parameter" }
+                $GetMailboxOutput = Get-Mailbox -Identity $Identity -Organization $Organization  -ErrorAction SilentlyContinue;
+            } else {
+                if ($debug) { Write-Host -ForegroundColor Green "Using -OrganizationalUnit parameter" }
+                $GetMailboxOutput = Get-Mailbox -Identity $Identity -OrganizationalUnit $Organization  -ErrorAction SilentlyContinue;
+            }
+        } else {
+            $GetMailboxOutput = Get-Mailbox -Identity $Identity -ErrorAction SilentlyContinue;
+        }
+
+        if (!$GetMailboxOutput) {
+            Write-Output "Unable to find $Identity in $Organization"
+            return $null
+        } else {
+            if ($debug) { Write-Verbose "Found [$($GetMailboxOutput.DisplayName)]" }
+        }
+
+        if (CheckForNoPIIAccess($script:GetMailboxOutput.DisplayName)) {
+            if ($debug) { Write-Host -ForegroundColor Magenta "No PII Access for [$Identity]" }
+        } else {
+            if ($debug) { Write-Verbose "Found [$($GetMailboxOutput.DisplayName)]" }
```

Comment:
Let's avoid using `$debug` here to determine what we want to display a simple `Write-Host` action. Use either `Write-Debug` or `Write-Verbose` instead. Besides, `$debug` would need to be set prior to running the script to have this code hit, which isn't the best way to handle code to be hit inside a script.